src/AutoRest.CSharp/MgmtTest/Generation/ResourceCollectionTestWriter.cs

```diff
@@ -276,30 +276,17 @@ protected void WriteMethodTest(MgmtClientOperation clientOperation, bool async)
                     }
 
                     WriteTestDecorator();
-                    if (clientOperation.IsPagingOperation(Context))
-                    {
-                        _writer.Append($"public void {testMethodName}{(exampleIdx > 0 ? (exampleIdx + 1).ToString() : "")}()");
-                    }
-                    else
-                    {
-                        _writer.Append($"public {GetAsyncKeyword(async)} {MgmtBaseTestWriter.GetTaskOrVoid(async)} {testMethodName}{(exampleIdx > 0 ? (exampleIdx + 1).ToString() : "")}()");
-                    }
+                    _writer.Append($"public {GetAsyncKeyword(async)} {MgmtBaseTestWriter.GetTaskOrVoid(async)} {testMethodName}{(exampleIdx > 0 ? (exampleIdx + 1).ToString() : "")}()");
                     using (_writer.Scope())
                     {
-                        _writer.LineRaw($"// Example: {exampleModel.Name}");
+                        _writer.Line($"// Example: {exampleModel.Name}");
                         clearVariableNames();
                         // WriteGetCollection(clientOperation, exampleModel, async);
                         WriteGetCollection(parentTp, operation.RequestPath.SerializedPath, exampleModel);
 
                         List<string> paramNames = WriteOperationParameters(methodParameters, new List<Parameter>(), exampleModel);
                         _writer.Line();
-                        _writer.Append($"{GetAwait(async && !clientOperation.IsPagingOperation(Context))} collection.{testMethodName}(");
-                        foreach (var paramName in paramNames)
-                        {
-                            _writer.Append($"{paramName},");
-                        }
-                        _writer.RemoveTrailingComma();
-                        _writer.LineRaw(");");
+                        WriteMethodTestInvocation(async, clientOperation.IsPagingOperation(Context)|| clientOperation.IsListOperation(Context, out var _), $"collection.{testMethodName}", paramNames);
```

Comment:
Could we change the approach of determining whether an operation is a paging operation? Here we are relying how the SDK generator is implemented (the methods you are invoking here (`IsPagingOperation` and `IsListOperation`) are `internal`).
Could we determine this by the return type of the operation?